runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3136,45 +3132,69 @@ remoteCompile(
-   std::vector<TR_OpaqueClassBlock*> unloadedClasses(compInfo->getUnloadedClassesTempList()->begin(), compInfo->getUnloadedClassesTempList()->end());
+   std::vector<TR_OpaqueClassBlock *> unloadedClasses(compInfo->getUnloadedClassesTempList()->begin(),
+                                                      compInfo->getUnloadedClassesTempList()->end());
    compInfo->getUnloadedClassesTempList()->clear();
-   std::vector<TR_OpaqueClassBlock*> illegalModificationList(compInfo->getIllegalFinalFieldModificationList()->begin(),
-                                                             compInfo->getIllegalFinalFieldModificationList()->end());
+   std::vector<TR_OpaqueClassBlock *> illegalModificationList(compInfo->getIllegalFinalFieldModificationList()->begin(),
+                                                              compInfo->getIllegalFinalFieldModificationList()->end());
    compInfo->getIllegalFinalFieldModificationList()->clear();
    // Collect and encode the CHTable updates; this will acquire CHTable mutex
-   auto table = (JITClientPersistentCHTable*)compInfo->getPersistentInfo()->getPersistentCHTable();
-   std::pair<std::string, std::string> chtableUpdates = table->serializeUpdates();
+   auto chTable = (JITClientPersistentCHTable *)persistentInfo->getPersistentCHTable();
+   std::pair<std::string, std::string> chtableUpdates = chTable->serializeUpdates();
    // Update the sequence number for these updates
    uint32_t seqNo = compInfo->incCompReqSeqNo();
    uint32_t lastCriticalSeqNo = !details.isJitDumpMethod() ? compInfo->getLastCriticalSeqNo() : 0;
    // If needed, update the seqNo of the last request that carried information that needed to be processed in order
-   if (!chtableUpdates.first.empty()
-       || !chtableUpdates.second.empty()
-       || !illegalModificationList.empty()
-       || !unloadedClasses.empty()
-       || details.isJitDumpMethod())
+   if (!chtableUpdates.first.empty() || !chtableUpdates.second.empty() ||
+       !illegalModificationList.empty() || !unloadedClasses.empty() || details.isJitDumpMethod())
+      {
       compInfo->setLastCriticalSeqNo(seqNo);
-   
+      }
    compInfo->getSequencingMonitor()->exit();
 
    uint32_t statusCode = compilationFailure;
    std::string codeCacheStr;
    std::string dataCacheStr;
    CHTableCommitData chTableData;
-   std::vector<TR_OpaqueClassBlock*> classesThatShouldNotBeNewlyExtended;
+   std::vector<TR_OpaqueClassBlock *> classesThatShouldNotBeNewlyExtended;
    std::string logFileStr;
    std::string svmSymbolToIdStr;
-   std::vector<TR_ResolvedJ9Method*> resolvedMirrorMethodsPersistIPInfo;
+   std::vector<TR_ResolvedJ9Method *> resolvedMirrorMethodsPersistIPInfo;
    TR_OptimizationPlan modifiedOptPlan;
    std::vector<SerializedRuntimeAssumption> serializedRuntimeAssumptions;
    std::vector<TR_OpaqueMethodBlock *> methodsRequiringTrampolines;
+   uint32_t methodIndex = (uint32_t)(method - clazz->ramMethods);
```

Comment:
Worth noting that this isn't going to be an index but a pointer diff. I think you have to divide by the size of the array element to get an index. Does the rest of the code assume it's an array index or a pointer diff?

Reply:
It is actually the index, and the rest of the code assumes that it's the index. I'm pretty sure this is standard C and C++, and it's true at least with gcc/g++. But I guess I can rewrite this as diff between pointers converted to `uint8_t *` divided by `sizeof` to make it less ambiguous.

Reply:
No I think if it's standard C/C++ then it's cleaner code as it stands.